internal/compiler-bridge/src/main/scala/xsbt/API.scala

```diff
@@ -210,7 +210,7 @@ final class API(val global: CallbackGlobal) extends Compat with GlobalHelpers wi
       !ignoredSymbol(sym) &&
       sym.isStatic &&
       !sym.isImplClass &&
-      !sym.hasFlag(Flags.JAVA) &&
+      (!sym.hasFlag(Flags.JAVA) || global.callback.isPickleJava) &&
```

Comment:
Would this fix the Scala source depending on Java from another subproject?
If so could you add a repro case similar to https://github.com/sbt/zinc/tree/develop/zinc/src/sbt-test/source-dependencies/subproject-pipelining?